authority/voting/server/state.go

```diff
@@ -1831,10 +1831,12 @@ func (s *state) backgroundFetchConsensus(epoch uint64) {
 	_, ok := s.documents[epoch]
 	if !ok {
 		go func() {
+			self := &config.Authority{Identifier: s.s.cfg.Server.Identifier, IdentityPublicKey: s.s.IdentityKey(), LinkPublicKey: s.s.linkKey.PublicKey(), Addresses: s.s.cfg.Server.Addresses}
+			allAuthorities := append(s.s.cfg.Authorities, self)
```

Comment:
what happens if it's already *in* the config, then we end up with two copies of ourself?
It seems like we should first check that we aren't in the config. Possibly we should just panic if the config is wrong?
Ie if we think we are an authority, but nobody else has a config that thinks that, trying to participate seems pointless.

Reply:
from client.New():
```
// New constructs a new pki.Client instance.
func New(cfg *Config) (pki.Client, error) {
        if cfg == nil {
                return nil, fmt.Errorf("voting/Client: cfg is mandatory")
        }
        if err := cfg.validate(); err != nil {
                return nil, err
        }

        c := new(Client)
        c.cfg = cfg 
        c.log = cfg.LogBackend.GetLogger("pki/voting/Client")
        c.pool = newConnector(cfg)
        c.verifiers = make([]cert.Verifier, len(c.cfg.Authorities))
        for i, auth := range c.cfg.Authorities {
                c.verifiers[i] = auth.IdentityPublicKey
        }
        c.threshold = len(c.verifiers)/2 + 1
        return c, nil
}
```

Reply:
if the authority were in the configuration twice, it would end up with the # of verifiers and threshold being wrong

Reply:
Right. But if we didn't magically add ourselves to the config, and instead mandated own pk being in the config, we could use the same config entry for all the auths. That seems a lot easier to manage/troubleshoot.

Reply:
so, duplicate the key material in the config from the Server section to the Authorities section?